core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -204,44 +224,46 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 throw new MapperParsingException("[" + propName + "] must not have a [null] value");
             }
             if (propName.equals("store")) {
-                builder.store(parseStore(propNode.toString()));
+                builder.store(nodeBooleanValue(name,"store", propNode.toString(), parserContext));
                 iterator.remove();
             } else if (propName.equals("index")) {
-                builder.index(parseIndex(name, propNode.toString()));
+                builder.index(nodeBooleanValue(name, "index", propNode, parserContext));
                 iterator.remove();
             } else if (propName.equals(DOC_VALUES)) {
-                builder.docValues(nodeBooleanValue(DOC_VALUES, propNode, parserContext));
+                builder.docValues(nodeBooleanValue(name, DOC_VALUES, propNode, parserContext));
                 iterator.remove();
             } else if (propName.equals("boost")) {
                 builder.boost(nodeFloatValue(propNode));
                 iterator.remove();
             } else if (parserContext.indexVersionCreated().before(Version.V_5_0_0_alpha1)
-                    && parseNorms(builder, propName, propNode, parserContext)) {
+                && parseNorms(builder, name, propName, propNode, parserContext)) {
                 iterator.remove();
             } else if (propName.equals("index_options")) {
                 builder.indexOptions(nodeIndexOptionValue(propNode));
                 iterator.remove();
             } else if (propName.equals("include_in_all")) {
                 if (parserContext.isWithinMultiField()) {
-                    throw new MapperParsingException("include_in_all in multi fields is not allowed. Found the include_in_all in field [" + name + "] which is within a multi field.");
+                    throw new MapperParsingException("include_in_all in multi fields is not allowed. Found the include_in_all in field ["
+                        + name + "] which is within a multi field.");
                 } else {
-                    builder.includeInAll(nodeBooleanValue("include_in_all", propNode, parserContext));
+                    builder.includeInAll(nodeBooleanValue(name, "include_in_all", propNode, parserContext));
                 }
                 iterator.remove();
             } else if (propName.equals("similarity")) {
                 SimilarityProvider similarityProvider = resolveSimilarity(parserContext, name, propNode.toString());
                 builder.similarity(similarityProvider);
                 iterator.remove();
             } else if (propName.equals("fielddata")
-                    && propNode instanceof Map
-                    && parserContext.indexVersionCreated().before(Version.V_5_0_0_alpha1)) {
+                && propNode instanceof Map
```

Comment:
Old indentation was better because it made it obvious that the conditions weren't part of the body.